test/asynchronous/test_client_bulk_write.py

```diff
@@ -384,15 +378,19 @@ async def test_returns_error_if_unacknowledged_too_large_insert(self):
             await client.bulk_write(models=models_replace, write_concern=WriteConcern(w=0))
 
     async def _setup_namespace_test_models(self):
-        max_message_size_bytes = (await async_client_context.hello)["maxMessageSizeBytes"]
-        max_bson_object_size = (await async_client_context.hello)["maxBsonObjectSize"]
+        # When compression is enabled, max_message_size is
+        # smaller to account for compression message header.
+        if async_client_context.client_options.get("compressors"):
+            max_message_size_bytes = self.max_message_size_bytes - 16
+        else:
+            max_message_size_bytes = self.max_message_size_bytes
```

Comment:
I had to update the `max_message_size_bytes` value for when compression is enabled.